raw_linux.go

```diff
@@ -338,3 +375,12 @@ func (s *sysSocket) SetTimeout(timeout time.Duration) error {
 	}
 	return unix.SetsockoptTimeval(s.fd, unix.SOL_SOCKET, unix.SO_RCVTIMEO, tv)
 }
+
+func rawHardwareAddr(addr net.HardwareAddr) (baddr [8]byte, err error) {
+	if len(addr) > 8 {
+		return baddr, fmt.Errorf("hwaddr '%s' is lager than 8 byte", addr)
+	}
```

Comment:
Are hardware addresses with less than 8 bytes valid?

Reply:
I don't believe so. But they don't cause a runtime error. Shall I change the check?